File: performance_tests/read_write/task.rb
require 'ro_crate'

num_crates = ARGV[0]
entities_in_crate = ARGV[1]
starting = Process.clock_gettime(Process::CLOCK_MONOTONIC)
for j in 1..num_crates.to_i do
  crate = ROCrate::Crate.new
  for i in 1..entities_in_crate.to_i do
    local_file = crate.add_file('./data/file'+i.to_s)
    person = ROCrate::Person.new(crate, '#joe'+i.to_s, {name: 'Joe Bloggs'})
    crate.add_contextual_entity(person)
    local_file.author = person
  end
  ROCrate::Writer.new(crate).write('crate'+j.to_s)
  parsed_crate = ROCrate::Reader.read('crate'+j.to_s)
end

ending = Process.clock_gettime(Process::CLOCK_MONOTONIC)

for j in 1..num_crates.to_i do
  FileUtils.remove_dir('crate'+j.to_s)
end

puts("crates = " + num_crates + ' ' + "entities in crate = " + entities_in_crate + " time = "+ (ending - starting).to_s)
File.open("mul_mix_rb.txt", "a") { |f| f.puts(ending - starting).to_s}

#ROCrate::Writer.new(crate).write('./an_ro_crate_directory')

File: performance_tests/read_write/task.py
from rocrate.rocrate import ROCrate
from rocrate.model.person import Person
from rocrate.model.file import File 
from rocrate.model.dataset import Dataset 

import time
import sys
import os
import shutil

def main(argv):
    num_crates = argv[0]
    times = argv[1]
    start = time.time()
    for j in range(int(num_crates)):
        crate = ROCrate()
        for i in range(int(times)):
            person = Person(crate, 'id'+str(i), {'name': 'Joe Bloggs'})
            file = crate.add_file('./data/file'+str(i), 'file'+str(i), fetch_remote = True)
            file['author'] = person
            crate.add(file)
            crate.add(person)
        crate.write('crate' + str(j))
        parsed_crate = ROCrate('crate' + str(j));
    end = time.time();
    print(num_crates + " " + str(end-start))
    with open('mul_mix_py.txt', 'a') as the_file:
        the_file.write(str(end-start)+'\n')
    for j in range(int(num_crates)):
            shutil.rmtree('crate' + str(j));

if __name__ == "__main__":
   main(sys.argv[1:])



